core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -581,7 +575,7 @@ private[zio] final class FiberContext[E, A](
             }
           } else {
             // Fiber was interrupted
-            curZio = ZIO.interrupt
+            curZio = ZIO.halt(state.get.interrupted)
```

Comment:
line info is lost in this change, use `haltWith` or `refailWithTrace` here?

Reply:
Fixed